### DISCUSSION TITLE
Access yup resolver context?

### DISCUSSION THREAD

**Discussion by Notaduck** [react-hook-form/react-hook-form#9084] [Q&A] [answered]
Hi There. I'm having some troubles with `react-hook-form` and `yup`. I need to pass in a dynmic value for a max number. I have tried to do it in the following way:

```js
 let dynamicValue = condition ? 1 : 2
  const {
    register,
    trigger,
    control,
    handleSubmit,
    watch,
    formState: { errors },
    getValues,
    setValue,
    reset,
  } = useForm({
    reValidateMode: "onChange",
    resolver: yupResolver(carDetailsSchema, {
      context: {
        maxMarketValue: dynamicValue,
      },
    }),
    shouldUnregister: true,
  });
```

```js
  marketValue: yup
    .number()
    .min(0)
    .when("$maxMarketValue", (maxValue, schema) => {
      console.log(maxValue);
      return schema.min(0).max(maxValue).required();
    }),
    ```
    
    However when I try to log the maxValue to the console is it undefind?

**Comment by bluebill1049** [comment]
not sure why it's not having the context, i can see the source code: https://github.com/react-hook-form/resolvers/blob/master/yup/src/yup.ts#L41-L82

**Comment by leapful** [comment, marked as answer]
@Notaduck @bluebill1049 :

As I remember, `context` will be provided via `useForm({ context: { // Some context props } })`, not from resolver schema options 😄 

The current implementation of yup resolver is using `Object.assign({}, schemaOptions, { context // this is the context from react-hook-form })` therefore the react-hook-form context always overrides the "context" from schemaOptions.

**Reply by bluebill1049** [reply]
oh wait, that's right i miss looked at the code above

**Reply by Notaduck** [reply]
Thank you so much, both of you. I can see it it the docs now as well: https://react-hook-form.com/api/useform

**Comment by itayperry** [comment]
Hi, does anyone know how to do the same in `zod`? 
How should I access the `react-form-hook` context from:
```typescript
useForm({
    context: {something: "something"},
    resolver: zodResolver(mySchema)
)
```
I can't find a way to do it. It seems to work only in `yup`.

**Reply by Athelian** [reply]
> Hi, does anyone know how to do the same in `zod`? How should I access the `react-form-hook` context from:
> 
> ```ts
> useForm({
>     context: {something: "something"},
>     resolver: zodResolver(mySchema)
> )
> ```
> 
> I can't find a way to do it. It seems to work only in `yup`.

@itayperry I did it like:

```ts
useForm<Inputs, Context>({
    resolver: (data, context, options) => {
      if (!context) {
        return {
          values: {},
          errors: {},
        }
      }

      return zodResolver(GET_VALIDATION(context.intl))(data, context, options)
    },
    context: {
      intl,
    },
})
```

`getValidation` return the schema

**Comment by eotkd4791** [comment]
@bluebill1049 
Hi, thank you for the discussion — I ran into the same behavior described here and looked into the internal implementation of yupResolver to better understand it.

Since this behavior can be unintuitive and is currently easy to miss, I was wondering if it would be acceptable to open an issue (and follow up with a PR) to update the react-hook-form/resolvers README. My intention would be to clearly document how context is handled so users can avoid unnecessary debugging and understand the correct usage upfront.

Please let me know if this sounds reasonable. I’d be happy to contribute the documentation update.

Thanks in advance.

**Reply by injae-kim** [reply]
+1 👍👍👍

**Reply by Han5991** [reply]
+2 👍👍👍

**Reply by BrendanC23** [reply]
This has been handled with [this issue](https://github.com/react-hook-form/resolvers/issues/834) and [this PR](https://github.com/react-hook-form/resolvers/pull/835). The PR has not been merged.
